### ISSUE TITLE
VS Code autocomplete imports from `#build/components` instead of `#components`

### ISSUE THREAD

**Issue by mrleblanc101** [nuxt/nuxt#30325]
### Environment

------------------------------
- Operating System: Darwin
- Node Version:     v20.18.0
- Nuxt Version:     3.14.1592
- CLI Version:      3.17.0
- Nitro Version:    2.10.4
- Package Manager:  npm@10.8.2
- Builder:          -
- User Config:      default
- Runtime Modules:  floating-vue/nuxt, @nuxtjs/color-mode@3.5.2, @vueuse/nuxt@12.0.0, @pinia/nuxt@0.9.0, @nuxt/icon@1.10.2, @nuxt/fonts@0.10.3, @nuxtjs/tailwindcss@6.12.2
- Build Modules:    -
------------------------------

### Reproduction

Neither StackBlitz or CodeSandbox seem to autocomplete my imports

### Describe the bug

I get this error when importing from `import { AppToast } from '#build/components';` instead of `import { AppToast } from '#components';`.

> [plugin:vite:import-analysis] Failed to resolve import "#build/components" from "app/plugins/vue3-toastify.ts". Does the file exist?

If I change the import manually, the app compile and everything works.
I'm using Nuxt 3 with v4 compatibility mode.

<img width="566" alt="Image" src="https://github.com/user-attachments/assets/de868d52-60a0-4ac5-91a2-72f8f6d5005d" />

### Logs

```shell-script

```

**Comment by mrleblanc101** [comment]
The doc says it's `#components`: https://nuxt.com/docs/guide/directory-structure/components#direct-imports

**Comment by danielroe** [comment]
This is not a bug but you can configure VSCode not to do this using `typescript.preferences.autoImportFileExcludePatterns`.

See https://github.com/posva/unplugin-vue-router/blob/main/.vscode/settings.json#L4-L6 for example.

A PR to add a sample `settings.json` to the docs (or maybe even the starter - though I'm reluctant to assume any particular IDE?) would be welcome. 🙏

**Comment by cernymatej** [comment]
Could an IDE selection be added to the init command, similar to how there is one for the package manager?

**Comment by mrleblanc101** [comment]
How is it not a bug if the app does not compile ?

**Comment by danielroe** [comment]
you should not  import from `#build/components` - it's a `.d.ts` declaration file for `#components`, not a file with any content.

**Comment by mrleblanc101** [comment]
I can't see that from VS Code autocomplete, so how would I know ?

**Comment by danielroe** [comment]
it's documented as `#components` - you should not import from `#build` at all. That's why I linked above how to tell VS Code not to import from a given path...

**Comment by mrleblanc101** [comment]
Is there no way to make it clearer ? 
I know nothing about the architecture of Nuxt? But maybe something like `#types/components` for all the types ?

**Comment by danielroe** [comment]
I'm very up for thinking about this - but this is a problem introduced by VS Code behaviour - you would never consider importing from that path if VS Code weren't suggesting it to you, I think.

Have you turned off auto-imports for components?

**Comment by mrleblanc101** [comment]
@danielroe No, but i am using Vue-final-modal and Vue3-toasting (with a custom component) so i need to pass a custom component in the `<script>` which requires me to manually import (i think) and that's when i stumbled upon this issue.

I think there is some discussion about moving away from auto-import (by default) so it might be a good idea to check both at the same time.

**Comment by danielroe** [comment]
great finding! yes, that would be fantastic 🙏

**Comment by mrleblanc101** [comment]
Woah, that would be a fantastic workaround until there is a way to make the import clearer either with clearer import name (ex: #types) or maybe there is something to be done on the IDE autocomplete suggestion side (maybe Volar could hint/provide more information in the VS Code Intellisense dropdown)
